src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/themes/AceThemes.java

```diff
@@ -71,7 +71,11 @@ private void applyTheme(Document document, final AceTheme theme)
       if (!currentUrl.equals(baseUrl) &&
          currentUrl.indexOf(baseUrl) == 0)
       {
-         pathUpCount = currentUrl.substring(baseUrl.length()).split("/").length;
+         String[] urlElements = currentUrl.substring(baseUrl.length()).split("/");
+         if (urlElements[urlElements.length - 1].contains("."))
```

Comment:
I don't think this is foolproof, since there's no rule against directory names having dots; in the general case, there is no way for us to tell by looking at a URL string whether the last segment refers to a file or a directory. 

Do we really need the `"../"` construction at all? Now that the theme URL handler is at a predictable location, it seems like we could build the theme URL off the `baseUrl`, regardless of the `currentUrl`. What do you think?

Reply:
Oh, and use an absolute link element rather than a relative one? That seems better to me, I'll test that.